x/curating/keeper/querier.go

```diff
@@ -37,5 +39,40 @@ func queryParams(ctx sdk.Context, k Keeper) ([]byte, error) {
 	return res, nil
 }
 
-// TODO: Add the modules query functions
-// They will be similar to the above one: queryParams()
+func queryPost(ctx sdk.Context, path []string, req abci.RequestQuery, k Keeper) ([]byte, error) {
+	i64, _ := strconv.ParseUint(path[0], 10, 32)
+	vendorID := uint32(i64)
+	postID := path[1]
+
+	post, _, err := k.GetPost(ctx, vendorID, postID)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
So this could return if its found or if there's an error. But I think returning one ErrPostNotFound is fine for both. Currently don't have custom errors but can create one for this. See https://github.com/cosmos/cosmos-sdk/blob/a63de303b8005a9143e696d8918bb4101d605a2c/x/staking/types/errors.go#L16 for example.